plasmapy/diagnostics/thomson.py

```diff
@@ -1,37 +1,286 @@
 """
-Defines the Thomson scattering analysis module as part of
-`plasmapy.diagnostics`.
+Defines the Thomson scattering analysis module as
+part of the diagnostics package.
 """
 
 __all__ = [
     "spectral_density",
+    "spectral_density_model",
 ]
 
 import astropy.constants as const
 import astropy.units as u
+import numbers
 import numpy as np
+import re
+import warnings
 
-from typing import List, Tuple, Union
+from lmfit import Model
+from typing import Any, Dict, List, Tuple, Union
 
 from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
 from plasmapy.formulary.frequencies import plasma_frequency
-from plasmapy.formulary.speeds import thermal_speed
-from plasmapy.particles import Particle
-from plasmapy.utils.decorators import validate_quantities
+from plasmapy.formulary.speeds import thermal_speed, thermal_speed_coefficients
+from plasmapy.particles import Particle, particle_mass
+from plasmapy.utils.decorators import (
+    bind_lite_func,
+    preserve_signature,
+    validate_quantities,
+)
+
+c_si_unitless = const.c.si.value
+e_si_unitless = const.e.si.value
+m_p_si_unitless = const.m_p.si.value
+m_e_si_unitless = const.m_e.si.value
+
 
 # TODO: interface for inputting a multi-species configuration could be
-# simplified using the plasmapy.classes.plasma_base class if that class
-# included ion and electron drift velocities and information about the ion
-# atomic species.
+#     simplified using the plasmapy.classes.plasma_base class if that class
+#     included ion and electron drift velocities and information about the ion
+#     atomic species.
+
+
+@preserve_signature
+def spectral_density_lite(
+    wavelengths,
+    probe_wavelength: numbers.Real,
+    n: numbers.Real,
+    Te: np.ndarray,
+    Ti: np.ndarray,
+    efract: np.ndarray = None,
+    ifract: np.ndarray = None,
+    ion_z: np.ndarray = None,
+    ion_mass: np.ndarray = None,
+    electron_vel: np.ndarray = None,
+    ion_vel: np.ndarray = None,
+    probe_vec: np.ndarray = None,
+    scatter_vec: np.ndarray = None,
+    instr_func_arr: np.ndarray = None,
+) -> Tuple[Union[np.floating, np.ndarray], np.ndarray]:
```

Comment:
```suggestion
def spectral_density_lite(
    wavelengths,
    probe_wavelength: numbers.Real,
    n: numbers.Real,
    Te: np.ndarray,
    Ti: np.ndarray,
    *,
    efract: np.ndarray,
    ifract: np.ndarray,
    ion_z: np.ndarray,
    ion_mass: np.ndarray,
    electron_vel: np.ndarray,
    ion_vel: np.ndarray,
    probe_vec: np.ndarray,
    scatter_vec: np.ndarray,
    instr_func_arr: np.ndarray = None,
) -> Tuple[Union[np.floating, np.ndarray], np.ndarray]:
```

Another thought, make everything a required keyword, except for possibly `instr_func_arr`.  `instr_func_arr` is the only real optional keyword in this list.  For any other function I'd prefer to see have the robustness provided by default values, but this is a lite-function and we're prioritizing speed.  So, doing this would eliminate all of the extra if-clauses.

Reply:
Yeah, good idea